### ISSUE TITLE
Proper RF impl investigation

### ISSUE THREAD

**Issue by mdymczyk** [h2oai/h2o4gpu#357]
First, check the current RF (XGBoost) accuracy against SKLearn.

Look into how long would it take to implement a proper RF.
